src/context/wishlist-context.js

```diff
@@ -0,0 +1,22 @@
+import { createContext, useContext, useReducer } from "react";
+import { wishlistReducer } from "reducer";
+
+const WishlistContext = createContext();
+
+const useWishlist = () => useContext(WishlistContext);
+
+const WishlistProvider = ({children}) => {
+
+const [wishlistState, wishlistDispatch ] = useReducer(wishlistReducer, {
+        wishlistProducts : [],
+        wishlistProductsCount:0
```

Comment:
I think there is no need of this, just an unnecessary code. To get length directly use wishlistProduct.length method